### PR TITLE
(fix, python): correct snippets for optional referenced requests when…

### PR DESCRIPTION
**Pull request by armandobelardo**
… inlined_request_params is True

### CODE REVIEW

File: generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py [hunk 1]
```diff
@@ -1107,13 +1109,19 @@ def generate_snippet(self) -> AST.Expression:
                 )
 
         if self.example.request is not None:
+            # For some reason the example type refernce is not marking it's type as optional, so we need to specify it so the
```
Comment (amckinney) [inline generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py:1112]: nit: 
```suggestion
            # For some reason the example type reference is not marking it's type as optional, so we need to specify it so the
```

File: generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py [hunk 2]
```diff
@@ -1107,13 +1109,19 @@ def generate_snippet(self) -> AST.Expression:
...
+            # snippets (and thus unit tests) write correctly
+            is_optional = False
+            if self.endpoint.request_body is not None and self.endpoint.request_body.get_as_union().type == "reference" and self.endpoint.request_body.get_as_union().request_body_type.get_as_union().type == "container" and self.endpoint.request_body.get_as_union().request_body_type.get_as_union().container.get_as_union().type == "optional":
+                is_optional = True
```
Comment (amckinney) [inline generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py:1116]: nit: Might be worthwhile to put this in a helper, unless we plan to remove this check for something better later (re: your comment).

**Review by amckinney — APPROVED** [review]
lgtm, only nits

File: generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py (hunk 2 above)
Comment (armandobelardo) [reply generators/python/src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py]: updated!

**Pull request merged**